types.go

```diff
@@ -15,54 +15,62 @@ import (
 
 const defaultGradientAbsTol = 1e-6
 
-// EvaluationType is used by a Method to specify the objective-function
-// information needed at an x location.
-type EvaluationType uint
+// Operation represents the set of operations requested by Method at each
+// iteration. It is a bitmap of *Iteration and *Evaluation constants.
```

Comment:
I thought that using * glob was clever, but I did not realize the connotation with pointer. I removed the * and reorganized the docs a bit.